CharlieBackend.Data/Repositories/Impl/Interfaces/IDashboardRepository.cs

```diff
@@ -0,0 +1,43 @@
+﻿using CharlieBackend.Core.DTO.Dashboard;
+using CharlieBackend.Core.DTO.StudentGroups;
+using CharlieBackend.Core.Entities;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Data.Repositories.Impl.Interfaces
+{
+    public interface IDashboardRepository
```

Comment:
Since you have a lot of the logic in this methods, we need to cover them with unit tests. Please research how we can mock DbSets with in memory SqlLite